controllers/core/openstackcontrolplane_controller.go

```diff
@@ -112,6 +112,8 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	// Remove/Reset the ready condition if exists
+	instance.Status.Conditions.Remove(condition.ReadyCondition)
```

Comment:
So I guess we need this as now ReadyCondition can go from True to False when a new service is enabled. I'm wondering that it would be better to move this to the deferred function where we set the ReadyCondition to True. So the manipulation of the ReadyCondition would be in one place. We either set that to true there if IsReady() true or to false if IsReady() is false.

Reply:
Currently ReadyCondition once true is left as True unless there is some error in the subsequent reconciliation. So this resets the condition at the beginning.